src/physics/reynolds_averaged_navier_stokes.cpp

```diff
@@ -325,6 +325,23 @@ ::max_convective_eigenvalue (const std::array<real,nstate> &conservative_soln) c
 }
 //----------------------------------------------------------------
 template <int dim, int nstate, typename real>
+real ReynoldsAveragedNavierStokesBase<dim,nstate,real>
+::max_convective_normal_eigenvalue (
+    const std::array<real,nstate> &conservative_soln,
+    const dealii::Tensor<1,dim,real> &normal) const
+{
+    const std::array<real,nstate_navier_stokes> conservative_soln_rans = extract_rans_conservative_solution(conservative_soln);
+
+    const dealii::Tensor<1,dim,real> vel = this->navier_stokes_physics->template compute_velocities<real>(conservative_soln_rans);
+
+    real vel_dot_n = 0.0;
+    for (int d=0;d<dim;++d) { vel_dot_n += vel[d]*normal[d]; };
+    const real max_normal_eig = sqrt(vel_dot_n*vel_dot_n);
```

Comment:
Should this have +c? I'm not sure whether it's intended to be the same as Euler or not.

Reply:
i asked him the same question back in PR #172 and apparently not, its consistent with his max conv eig function

Reply:
Mmmh, kinda agree with @cpethrick on this one. Why would Euler have "+c", but not NS. What if we're doing low-Mach cases... then +c becomes the dominant term.

Reply:
@cyanmcg

Reply:
For RANS, the max eigenvalues based on its flux Jacobian should be u since their convective fluxes always take the form of u_x*variable_turbulence (or u_y*variable_turbulence  or u_z*variable_turbulence), so the wave that carries this turbulent characteristic is the one with the flow velocity. And all other terms are considered as in the source, which does not contribute to RANS fluxes.

Reply:
@dougshidong the RANS class in the code only describes the additional PDE(s) for RANS turbulence models, which is derived from the model class, the RANS part (here I think the word using may lead confusion, I mean RANS in our code is just for the RANS turbulence model but not Reynolds-averaged Navier-Stokes + turbulence model) still using describe by the NS in the code, and as I last checked its function inherits from Euler should return the max convective eigenvalue as u+c. The RANS turbulence model only contributes to the dissipative flux, I think it may need some change to consider this part in the dissipative flux but not the convective flux

Reply:
Makes a lot of sense.